server/server.go

```diff
@@ -15,15 +15,78 @@
 package server
 
 import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
 	framework "github.com/sgnl-ai/adapter-framework"
 	api_adapter_v1 "github.com/sgnl-ai/adapter-framework/api/adapter/v1"
 	"github.com/sgnl-ai/adapter-framework/server/internal"
 )
 
 // New returns an AdapterServer that wraps the given high-level
-// Adapter implementation.
-func New[Config any](adapter framework.Adapter[Config]) api_adapter_v1.AdapterServer {
-	return &internal.Server[Config]{
-		Adapter: adapter,
+// Adapter implementation with the Tokens field populated.
+func New[Config any](adapters map[string]framework.Adapter[Config]) api_adapter_v1.AdapterServer {
+	path, exists := os.LookupEnv("AUTH_TOKENS_PATH")
+	if !exists {
+		panic("AUTH_TOKENS_PATH environment variable not set")
+	}
+
+	server := &internal.Server[Config]{
+		Adapters: adapters,
+		Tokens:   getTokensFromPath(path),
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		panic(fmt.Sprintf("failed to create file watcher: %s", err.Error()))
+	}
```

Comment:
In order to not miss any update, change the order of operations here:
1. Create the watcher.
2. Add the patch to the watcher.
3. Call `getTokensFromPath` and set the initial value of `Tokens`.
4. Start the goroutine and start consuming watcher events.